### PR TITLE
Fix and Enable Most ETW EventSource Tests

### LINKED ISSUE #4864: Vaildation of ETW events need TraceEvent
EventSource needs TraceEvent to validate ETW events in the tests.

### PR DESCRIPTION
**Pull request by brianrob**
This change bumps the version of TraceEvent to 2.0.4 and enables almost all of the ETW EventSource tests, with small fixes for several of them.

System.Diagnostics.Tracing.Tests is also OS-specific now since it references ETW-only code.

Fixes #4864.

### CODE REVIEW

File: src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsWriteEvent.cs [hunk 1]
```diff
@@ -647,6 +652,8 @@ private void Test_WriteEvent_ByteArray(bool useSelfDescribingEvents, Listener li
                         Assert.Equal(1000, (long)evt.PayloadValue(1, "lng"));
                     }));
 
+                /* TODO: NULL byte array does not seem to be supported.
+                 * An EventSourceMessage event is written for this case.
```
Comment (stephentoub) [inline src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsWriteEvent.cs]: Is this https://github.com/dotnet/coreclr/pull/1988?

File: src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/FuzzyTests.cs [hunk 2]
```diff
@@ -13,7 +13,7 @@
 using System.Diagnostics.Tracing;
 #endif
 using Xunit;
-#if USE_ETW // TODO: Enable when TraceEvent is available on CoreCLR. GitHub issue #4864.
+#if USE_ETW
```
Comment (stephentoub) [inline src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/FuzzyTests.cs]: Do we need a compilation constant for this?  If the issue is just whether to run certain tests on Windows, can we use `[PlatformSpecific(TestPlatforms.Windows)]`?

File: src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsWriteEvent.cs (hunk 1 above)
Comment (brianrob) [reply src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsWriteEvent.cs]: Close, but it doesn't look like this matches one of the overloads that was modified in that PR.  I suspect we need to modify the params overload to account for this one.

File: src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/FuzzyTests.cs (hunk 2 above)
Comment (brianrob) [reply src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/FuzzyTests.cs]: We may be able to get rid of this, but I'm not 100% sure.  Filed #27116.

**Pull request merged**